src/lib/eliom_uri.shared.ml

```diff
@@ -536,21 +556,22 @@ let make_post_uri_components_
 
     let ssl = Eliom_request_info.get_csp_ssl_sp sp in
     let https = is_https https ssl service in
-    let absolute =
+    let absolute' =
       if absolute || https <> ssl
       then Some (make_proto_prefix ?hostname ?port https)
       else if absolute_path
       then Some "/"
       else None
     in
 
-
     (* absolute URL does not work behind a reverse proxy! *)
     let uri =
-      match absolute with
-      | Some proto_prefix ->
+      match absolute', !app_path with
+      | Some proto_prefix, Some app_path when absolute ->
```

Comment:
We should use this kind of URL only if the URL is generated client-side, by a client app.

if I understand well your code, app_path is set only by client_apps, so it's ok. But should be explained in comment.

Also please explain in comment why we are doing that.